lib/ansible/modules/clustering/k8s.py

```diff
@@ -0,0 +1,529 @@
+    name: testing
+    api_version: v1
+    kind: namespace
+    state: present
+
+- name: Create an OpenShift project
+  k8s:
+    name: testing
+    api_version: v1
+    kind: project
+    state: present
+
+- name: Create a Service object from an inline definition
+  k8s:
+    state: present
+    resource_definition:
+      apiVersion: v1
+      kind: Service
+      metadata:
+        name: web
+        namespace: testing
+        labels:
+          app: galaxy
+          service: web
+      spec:
+        selector:
+          app: galaxy
+          service: web
+        ports:
+        - protocol: TCP
+          targetPort: 8000
+          name: port-8000-tcp
+          port: 8000
+
+- name: Create a Service object by reading the definition from a file
+  k8s:
+    state: present
+    src: /testing/service.yml
+    
+- name: Remove an existing Service object
+  k8s:
+    state: absent
+    api_version: v1
+    kind: service
+    namesapce: testing
+    name: web
+'''
+
+RETURN = '''
+result:
+  description:
+    The object that was created, patched, or otherwise present. In the case of a deletion, will be empty.
+  returned: success
+  type: dict 
+...
+'''
+
+class K8sException(Exception):
+    pass
+
+
+class K8sModule(AnsibleModule):
+
+    def __init__(self, argspec=None, mutually_exclusive=None, supports_check_mode=False):
+
+        if not HAS_K8S_MODULE_HELPER:
+            raise K8sException(
+                "This module requires the OpenShift Python client. Try `pip install openshift`"
+            )
+
+        if not HAS_YAML:
+            raise K8sException(
+                "This module requires PyYAML. Try `pip install PyYAML`"
+            )
+
+        AnsibleModule.__init__(self,
+                               argument_spec=argspec,
+                               supports_check_mode=supports_check_mode,
+                               mutually_exclusive=mutually_exclusive)
+
+        self.kind = self.params.pop('kind')
+        self.api_version = self.params.pop('api_version')
+        self.resource_definition = self.params.pop('resource_definition')
+        self.src = self.params.pop('src')
+        if self.src:
+            self.resource_definition = self.load_resource_definition(self.src)
+
+        if self.resource_definition:
+            self.api_version = self.resource_definition.get('apiVersion')
+            self.kind = self.resource_definition.get('kind')
+
+        self.api_version = self.api_version.lower()
+        self.kind = self._to_snake(self.kind)
+
+        try:
+            self.helper = self._get_helper(self.api_version, self.kind)
+        except Exception as exc:
+            raise K8sException(
+                "Error initializing AnsibleModuleHelper: {}".format(str(exc))
+            )
+
+    @staticmethod
+    def _get_helper(api_version, kind):
+        try:
+            helper = KubernetesAnsibleModuleHelper(api_version, kind)
+            helper.get_model(api_version, kind)
+        except:
```

Comment:
Bare excepts scare me. Also, there are resource names which appear in both openshift and kubernetes, and I think we need to provide a way for the user to determine which project's API they want to use. This approach could be a sane default though.